### PR TITLE
Refactor pose estimator

### PR DESCRIPTION
**Pull request by patapufate**
This patch:

- Remove unnecessary boilerplate in CMakeList.txt and Package.xml.
- Remove commented/obsolete code
- Add mantainer and license.
- Add Roslint and fix linter errors.
- Change folder structure for header files. Now implemented as "include/<package_name>/header_file.h".
- Remove obsolete code
- Separate Header and Source file

It needs thorough testing, as I don't have the hardware at hand.

### CODE REVIEW

File: pose_estimator/src/pose_estimator.cpp [hunk 1]
```diff
@@ -32,361 +32,321 @@
...
 
-  void gpsDataCallback(const sensor_msgs::NavSatFix data) {
-    if (data.status.status >= 0) {
-      boost::mutex::scoped_lock lock(m_mutDataLock);
-      cur_latitude = data.latitude;
-      cur_longitude = data.longitude;
-      cur_gps_time = data.header.stamp.toSec();
-    }
-  }
+void PoseEstimatorNode::ahrsDataCallback(const sensor_msgs::Imu data)
```
Comment (hidmic) [inline pose_estimator/src/pose_estimator.cpp:256]: @patapufate nit: make it a const ref
```suggestion
void PoseEstimatorNode::ahrsDataCallback(const sensor_msgs::Imu& data)
```

File: pose_estimator/src/pose_estimator.cpp [hunk 2]
```diff
@@ -32,361 +32,321 @@
...
+  cur_rpy_ang[CorrectedData::PITCH] = pitch * -1.0;
+  cur_rpy_ang[CorrectedData::YAW] = yaw;
+  ahrs_ok = true;
+}
 
-  void rpmDataCallback(const thruster_control::ReportRPM data) {
-    boost::mutex::scoped_lock lock(m_mutDataLock);
-    cur_speed = (double)data.rpms / rpmPerKnot;
-  }
+void PoseEstimatorNode::rpmDataCallback(const thruster_control::ReportRPM data)
```
Comment (hidmic) [inline pose_estimator/src/pose_estimator.cpp:270]: @patapufate nit: make it a const ref
```suggestion
void PoseEstimatorNode::rpmDataCallback(const thruster_control::ReportRPM& data)
```

File: pose_estimator/src/pose_estimator.cpp [hunk 3]
```diff
@@ -32,361 +32,321 @@
...
+{
+  boost::mutex::scoped_lock lock(m_mutDataLock);
+  cur_speed = static_cast<double>(data.rpms) / rpmPerKnot;
+}
 
-  void pressureDataCallback(const sensor_msgs::FluidPressure data) {
+void PoseEstimatorNode::gpsDataCallback(const sensor_msgs::NavSatFix data)
```
Comment (hidmic) [inline pose_estimator/src/pose_estimator.cpp:276]: @patapufate nit: make it a const ref
```suggestion
void PoseEstimatorNode::gpsDataCallback(const sensor_msgs::NavSatFix& data)
```

File: pose_estimator/src/pose_estimator.cpp [hunk 4]
```diff
@@ -32,361 +32,321 @@
...
   }
+}
 
-  void processDynPos() {
-    boost::mutex::scoped_lock data_lock(m_mutDataLock);
-    m_correctedData.rpy_ang.x = cur_rpy_ang[CorrectedData::ROLL];
-    m_correctedData.rpy_ang.y = cur_rpy_ang[CorrectedData::PITCH];
-    m_correctedData.rpy_ang.z = cur_rpy_ang[CorrectedData::YAW];
-  }
+void PoseEstimatorNode::pressureDataCallback(const sensor_msgs::FluidPressure data)
```
Comment (hidmic) [inline pose_estimator/src/pose_estimator.cpp:287]: @patapufate nit: make it a const ref 
```suggestion
void PoseEstimatorNode::pressureDataCallback(const sensor_msgs::FluidPressure& data)
```

**Review by hidmic — APPROVED** [review]
LGTM but for a few minor comments.

File: pose_estimator/src/pose_estimator.cpp (hunk 1 above)
Comment (patapufate) [reply pose_estimator/src/pose_estimator.cpp]: fixed

File: pose_estimator/src/pose_estimator.cpp (hunk 2 above)
Comment (patapufate) [reply pose_estimator/src/pose_estimator.cpp]: fixed

File: pose_estimator/src/pose_estimator.cpp (hunk 3 above)
Comment (patapufate) [reply pose_estimator/src/pose_estimator.cpp]: fixed

File: pose_estimator/src/pose_estimator.cpp (hunk 4 above)
Comment (patapufate) [reply pose_estimator/src/pose_estimator.cpp]: fixed

**Pull request merged**